sg_PinkNoiseGenerator.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+ This file is part of SpatGRIS.
+
+ Developers: Gaël Lane Lépine, Samuel Béland, Olivier Bélanger, Nicolas Masson
+
+ SpatGRIS is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ SpatGRIS is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with SpatGRIS.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "sg_PinkNoiseGenerator.hpp"
+
+#include "Data/StrongTypes/sg_Dbfs.hpp"
+#include "Data/sg_Narrow.hpp"
+
+namespace
+{
+float pinkNoiseC0{};
+float pinkNoiseC1{};
+float pinkNoiseC2{};
+float pinkNoiseC3{};
+float pinkNoiseC4{};
+float pinkNoiseC5{};
+float pinkNoiseC6{};
+
+} // namespace
+
+namespace gris
+{
+//==============================================================================
+void fillWithPinkNoise(float * const * samples, int const numSamples, int const numChannels, float const gain)
+{
+    static constexpr auto FAC{ 1.0f / (static_cast<float>(RAND_MAX) / 2.0f) };
+    static constexpr dbfs_t CORRECTION_DB{ -18.2f };
+
+    static auto const CORRECTION{ CORRECTION_DB.toGain() };
+
+    for (int sampleIndex{}; sampleIndex < numSamples; ++sampleIndex) {
+        auto const rnd{ narrow<float>(rand()) * FAC - 1.0f };
```

Comment:
I'd propose to use a proper C++ random device for this, which will avoid potential mutex locks / unlocks in rand() if this function ever ends up being called in an audio thread: 

```
// in the anonymous namespace on top
std::random_device rd; 
std::mt19937_64 gen(rd()); // or maybe std::minstd_rand is more than enough
std::uniform_real_distribution<float> dist(-1, 1.); // if i read the auto const rnd line correctly

// in fillWithPinkNoise
narrow<float>(dist(random_engine))
```